crates/redis/src/lib.rs

```diff
@@ -52,30 +55,81 @@ impl TriggerExecutor for RedisTrigger {
     type RunConfig = NoArgs;
 
     async fn new(engine: TriggerAppEngine<Self>) -> Result<Self> {
-        let address = engine
+        let default_address: String = engine
             .trigger_metadata::<TriggerMetadata>()?
             .unwrap_or_default()
             .address;
 
+        let mut addresses = vec![default_address.clone()];
+
         let mut channel_components: HashMap<String, Vec<String>> = HashMap::new();
 
         for (_, config) in engine.trigger_configs() {
+            let address = config.address.clone().unwrap_or(default_address.clone());
+            if !addresses.contains(&address) {
+                addresses.push(address.clone());
+            }
             channel_components
-                .entry(config.channel.clone())
+                .entry(format!("{}:{}", address, config.channel.clone()))
                 .or_default()
                 .push(config.component.clone());
         }
         Ok(Self {
-            engine,
-            address,
+            engine: Arc::new(engine),
+            addresses,
             channel_components,
         })
     }
 
     /// Run the Redis trigger indefinitely.
     async fn run(self, _config: Self::RunConfig) -> Result<()> {
-        let address = &self.address;
+        let addresses = self.addresses.clone();
+        let mut tasks = Vec::new();
+        // Spawn an asynchronous task for each iteration
+        for address in addresses {
+            let trigger = self.clone();
+            // Spawn an asynchronous task with a closure capturing self by reference
+            println!("Starting listener for {address}");
+            tasks.push(tokio::spawn(
```

Comment:
This is a style thing, but throughout this PR you show a strong preference for loops pushing onto mutable vectors, rather than using iterator functions - is that deliberate?  It's not wrong, and at this scale mostly a style thing, I'm just curious.

Reply:
Just a force of habit. Will update to use an iterator.